_assets/js/carousel.js

```diff
@@ -0,0 +1,335 @@
+/*!
+ * Adapted from W3C WAI’s Web Accessibility Tutorials carousel
+ * https://github.com/w3c/wai-tutorials/
+ */
+
+const mobileCarousel = () => {
+  // Initial constiables
+  let carousel;
+  let slides;
+  let index;
+  let slidenav;
+  let settings;
+  let timer;
+  let setFocus;
+  let animationSuspended;
+
+  // Helper function: Iterates over an array of elements
+  function forEachElement(elements, fn) {
+    for (let i = 0; i < elements.length; i++) fn(elements[i], i);
+  }
+
+  // Helper function: Remove Class
+  function removeClass(el, className) {
+    if (el.classList) {
+      el.classList.remove(className);
+    } else {
+      el.className = el.className.replace(
+        new RegExp('(^|\\b)' + className.split(' ').join('|') + '(\\b|$)', 'gi'),
+        ' '
+      );
+    }
+  }
+
+  // Helper function: Test if element has a specific class
+  function hasClass(el, className) {
+    if (el.classList) {
+      return el.classList.contains(className);
+    } else {
+      return new RegExp('(^| )' + className + '( |$)', 'gi').test(el.className);
+    }
+  }
+
+  // Initialization for the carousel
+  // Argument: set = an object of settings
+  // Possible settings:
+  // id <string> ID of the carousel wrapper element (required).
+  // slidenav <bool> If true, a list of slides is shown.
+  // animate <bool> If true, the slides can be animated.
+  // startAnimated <bool> If true, the animation begins
+  // immediately.
+  // If false, the animation needs
+  // to be initiated by clicking
+  // the play button.
+  function init(set) {
```

Comment:
this function is very long. It'd be nice to have it broken out into meaningful helper functions, such as:

function init(set) {
  initControls(set);
  initSlidenav(set);
  ...
}